frontend/src/metabase/query_builder/components/DimensionList.jsx

```diff
@@ -185,17 +186,21 @@ export default class DimensionList extends Component {
     } = this.props;
     const dimension = useOriginalDimension
       ? item.dimension
-      : item.dimension.defaultDimension() || item.dimension;
+      : // XXX: 7. Important note that dimension with binning options is `dimension.defaultDimension()`
+        item.dimension.defaultDimension() || item.dimension;
     const shouldExcludeBinning =
-      !enableSubDimensions &&
-      !useOriginalDimension &&
-      dimension instanceof FieldDimension &&
-      dimension.binningStrategy();
-
-    if (
-      shouldExcludeBinning ||
-      (preventNumberSubDimensions && dimension.field().isSummable())
-    ) {
+      (!enableSubDimensions &&
+        !useOriginalDimension &&
+        dimension instanceof FieldDimension &&
+        dimension.binningStrategy()) ||
+      (preventNumberSubDimensions && dimension.field().isSummable()) ||
+      dimension.field().isFK();
+    // XXX 14. 🎉 dimension.field() after clicking to select each dimension in the `<DimensionList />`
+    // 1. Reviews join native Orders question: this._createFallbackField(), field().isSummable() is `true`
+    // 2. Orders join native Orders question: this._findMatchingQueryField(), field().isSummable() is `false` because it's a FK
+    // 3. Native Orders question join Orders this._findMatchingQueryField(), field().isSummable() is `true`
+
+    if (shouldExcludeBinning) {
```

Comment:
I moved the extra condition inside `shouldExcludeBinning` since it sounds weird to use both `shouldExludeBinnig` together with extra conditions. I think the whole block should just refer to the variable `shouldExcludeBinning` alone.